File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n = nums.size();
        unordered_map<int, int> mp;
        vector<vector<int>> ans;
        for (auto& a : nums) {
            mp[a]++;
        }

        for (auto& p : mp) {
            int freq = p.second;

            ans.push_back({freq, p.first});
        }

        sort(ans.begin(), ans.end(), greater<vector<int>>());
        return ans[0][1];
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer,Integer>mp=new HashMap<>();
        int n=nums.length;

        for(int i=0;i<nums.length;++i){
            mp.put(nums[i],mp.getOrDefault(nums[i],0)+1);
        }

        for(int i=0;i<nums.length;++i){
            if(mp.get(nums[i])>n/2){
                return nums[i];
            }
        }
        return 0;
    }
}